src/Sentry.Unity/UnityEventProcessor.cs

```diff
@@ -6,66 +6,80 @@
 
 namespace Sentry.Unity
 {
-    internal class UnityEventProcessor : ISentryEventProcessor
+    internal class UnityEventProcessor :
+        ISentryEventProcessor,
+        ISentryTransactionProcessor
     {
         private readonly SentryUnityOptions _sentryOptions;
         private readonly MainThreadData _mainThreadData;
 
-
         public UnityEventProcessor(SentryUnityOptions sentryOptions, SentryMonoBehaviour sentryMonoBehaviour)
         {
             _sentryOptions = sentryOptions;
             _mainThreadData = sentryMonoBehaviour.MainThreadData;
         }
 
+        public Transaction Process(Transaction transaction)
+        {
+            SetEventContext(transaction);
+            return transaction;
+        }
+
         public SentryEvent Process(SentryEvent @event)
+        {
+            SetEventContext(@event);
+
+            @event.ServerName = null;
+
+            return @event;
+        }
+
+        private void SetEventContext(IEventLike sentryEvent)
         {
             try
             {
-                PopulateDevice(@event.Contexts.Device);
+                PopulateDevice(sentryEvent.Contexts.Device);
                 // Populating the SDK Integrations here (for now) instead of UnityScopeIntegration because it cannot be guaranteed
                 // that it got added last or that there was not an integration added at a later point
-                PopulateSdkIntegrations(@event.Sdk);
+                PopulateSdkIntegrations(sentryEvent.Sdk);
                 // TODO revisit which tags we should be adding by default
-                @event.SetTag("unity.is_main_thread", _mainThreadData.IsMainThread().ToTagValue());
+                sentryEvent.SetTag("unity.is_main_thread", _mainThreadData.IsMainThread().ToTagValue());
             }
-            catch (Exception ex)
+            catch (Exception exception)
             {
-                _sentryOptions.DiagnosticLogger?.LogError("{0} processing failed.", ex, nameof(SentryEvent));
+                _sentryOptions.DiagnosticLogger?.LogError("{0} processing failed.", exception, nameof(SentryEvent));
             }
-
-            @event.ServerName = null;
-
-            return @event;
         }
 
         private void PopulateDevice(Device device)
         {
-            if (_mainThreadData.IsMainThread())
+            if (!_mainThreadData.IsMainThread())
```

Comment:
Just to confirm: This is just a refactor? Or am I missing a change here I can't spot?

Reply:
I inverted so as to have an early exit and make it easier to reason about